sympy/tensor/tests/test_tensor_operators.py

```diff
@@ -117,3 +123,183 @@ def test_replace_arrays_partial_derivative():
+        PartialDerivative(A(i), D(j))\
+        + PartialDerivative(B(i), D(j))\
+        + PartialDerivative(C(i), D(j))
+
+    # check sum rule for D(tensor, H(j, k))
+    expr1ca = PartialDerivative(A(i), H(j, k))
+    assert expr1ca._expand_partial_derivative() ==\
+        PartialDerivative(A(i), H(j, k))
+    expr1cb = PartialDerivative(A(i) + B(i), H(j, k))
+    assert (expr1cb._expand_partial_derivative() ==
+            PartialDerivative(A(i), H(j, k))
+            + PartialDerivative(B(i), H(j, k)))
+    expr1cc = PartialDerivative(A(i) + B(i) + C(i), H(j, k))
+    assert (expr1cc._expand_partial_derivative() ==
+            PartialDerivative(A(i), H(j, k))
+            + PartialDerivative(B(i), H(j, k))
+            + PartialDerivative(C(i), H(j, k)))
+
+    # check sum rule for D(D(tensor, D(j)), H(k, m))
+    expr1da = PartialDerivative(A(i), (D(j), H(k, m)))
+    assert expr1da._expand_partial_derivative() ==\
+        PartialDerivative(A(i), (D(j), H(k, m)))
+    expr1db = PartialDerivative(A(i) + B(i), (D(j), H(k, m)))
+    assert expr1db._expand_partial_derivative() ==\
+        PartialDerivative(A(i), (D(j), H(k, m)))\
+        + PartialDerivative(B(i), (D(j), H(k, m)))
+    expr1dc = PartialDerivative(A(i) + B(i) + C(i), (D(j), H(k, m)))
+    assert expr1dc._expand_partial_derivative() ==\
+        PartialDerivative(A(i), (D(j), H(k, m)))\
+        + PartialDerivative(B(i), (D(j), H(k, m)))\
+        + PartialDerivative(C(i), (D(j), H(k, m)))
+
+
+def test_expand_partial_derivative_constant_factor_rule():
+    for count in range(10):
+        pos_random_int1 = sympify(randint(0, 1000))
+        pos_random_int2 = sympify(randint(0, 1000))
+        neg_random_int = sympify(randint(-1000, -1))
+
+        c1 = Rational(pos_random_int1, pos_random_int2)
+        c2 = Rational(neg_random_int, pos_random_int2)
+        c3 = Rational(pos_random_int1, neg_random_int)
+
+        expr2a = PartialDerivative(pos_random_int1*A(i), D(j))
+        assert expr2a._expand_partial_derivative() ==\
+            pos_random_int1*PartialDerivative(A(i), D(j))
+
+        expr2b = PartialDerivative(neg_random_int*A(i), D(j))
+        assert expr2b._expand_partial_derivative() ==\
+            neg_random_int*PartialDerivative(A(i), D(j))
+
+        expr2ca = PartialDerivative(c1*A(i), D(j))
+        assert expr2ca._expand_partial_derivative() ==\
+            c1*PartialDerivative(A(i), D(j))
+
+        expr2cb = PartialDerivative(c2*A(i), D(j))
+        assert expr2cb._expand_partial_derivative() ==\
+            c2*PartialDerivative(A(i), D(j))
+
+        expr2cc = PartialDerivative(c3*A(i), D(j))
+        assert expr2cc._expand_partial_derivative() ==\
+            c3*PartialDerivative(A(i), D(j))
+
+
+def test_expand_partial_derivative_full_linearity():
+    for count in range(10):
+        pos_random_int1 = sympify(randint(0, 1000))
+        pos_random_int2 = sympify(randint(0, 1000))
+        neg_random_int = sympify(randint(-1000, -1))
```

Comment:
Maybe it's a bit exaggerated to test the same test 10 times... the random integers could be created only once.

Reply:
I wanted to avoid that there are special cases appearing, which lead to a passing test one times and do not pass the next time. But yes, maybe 10 is exaggeration. :-)

Reply:
Array operations can be tested like that, but I think it's not really necessary with symbolic expressions.